yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorDelegationTokenUpdater.scala

```diff
@@ -76,7 +76,10 @@ private[spark] class ExecutorDelegationTokenUpdater(
         SparkHadoopUtil.get.getTimeFromNowToRenewal(
           sparkConf, 0.8, UserGroupInformation.getCurrentUser.getCredentials)
       if (timeFromNowToRenewal <= 0) {
-        executorUpdaterRunnable.run()
+        // If the `timeFromNowToRenewal` equals 0, it's better to wait 1 minutes to schedule for
+        // avoid cycle calling and cause StackOverflow Exception.
```

Comment:
We end up scheduling the next run of the `executorUpdaterRunnable` from within this class. If the executor renewal interval is set to 0, we end up calling it again from within the same thread (indefinitely). But I am not sure why it would be 0 here though (since this is started only if the login is from keytab).